lib/gather/local-gather.js

```diff
@@ -0,0 +1,198 @@
+module.exports = function parseMeta(meta = {}) {
```

Comment:
I think eventually we will want to deprecate folks from using this (e.g. the native-class codemod should bring this function back inline), but this seems like a good middle step.

Lets make a few tweaks:

* rename it to `lib/gather/analyze-ember-object.js` (thats basically what it does AFAICT)
* move the `.proto()` stuff into here
* change the argument it receives to be the export to analyze and modify the function to do something like:

```suggestion
module.exports = function analyzeEmberObject(possibleEmberObject) {
  if (typeof possibleEmberObject !== 'object' || possibleEmberObject === null) {
    return undefined;
  }

  if (typeof possibleEmberObject.proto !== 'function') {
    return undefined;
  }
  
  let proto = possibleEmberObject.proto();
  let meta = parseMeta(Ember.meta(proto));
```

Reply:
Sounds good.